pkg/tasks/cleanup_trashcans_service.go

```diff
@@ -0,0 +1,81 @@
+package tasks
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/ottogroup/penelope/pkg/http/impersonate"
+	"github.com/ottogroup/penelope/pkg/repository"
+	"github.com/ottogroup/penelope/pkg/secret"
+	"github.com/ottogroup/penelope/pkg/service/gcs"
+	"go.opencensus.io/trace"
+	"time"
+)
+
+func newCleanupTrashcansService(ctxIn context.Context, tokenSourceProvider impersonate.TargetPrincipalForProjectProvider, credentialsProvider secret.SecretProvider) (*cleanupTrashcansService, error) {
+	ctx, span := trace.StartSpan(ctxIn, "newPrepareBackupJobsService")
+	defer span.End()
+
+	backupRepository, err := repository.NewBackupRepository(ctx, credentialsProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cleanupTrashcansService{tokenSourceProvider: tokenSourceProvider, backupRepository: backupRepository}, nil
+}
+
+type cleanupTrashcansService struct {
+	tokenSourceProvider impersonate.TargetPrincipalForProjectProvider
+	backupRepository    repository.BackupRepository
+}
+
+func (s *cleanupTrashcansService) Run(ctxIn context.Context) {
+	ctx, span := trace.StartSpan(ctxIn, "(*cleanupTrashcansService).Run")
+	defer span.End()
+
+	backups, err := s.backupRepository.GetBackupsByCleanupTrashcanStatus(ctx, repository.ScheduledTrashcanCleanupStatus)
```

Comment:
what happen, if the cronjob triggers, but the preceding task isn't done? Does app engine ensure, that is cannot run concurrently? If not, you need to change the CleanupStatus, as soon the job is grabbed to something like "Running"

Reply:
using new state Running to display running jobs.